server/src/lib.rs

```diff
@@ -218,7 +218,22 @@ fn fetch_iframely_and_pictshare_data(
         None
       }
     },
-    None => None,
+
+    None => match url {
+      Some(url) => match fetch_pictshare(&url) {
+        // Try to generate a small thumbnail if iframely is not supported
+        Ok(res) => {
+          let mut split_url: Vec<&str> = res.url.split('/').collect();
+          split_url.insert(split_url.len() - 1, "192");
+          Some(split_url.join("/"))
+        }
```

Comment:
Don't store the hardcoded thumbnail in the table like that, just use pictshare's url and the front end needs some work to check if its a pictshare url. This is in `utils.tsx` and `post-listing.tsx`

Reply:
Sounds good. I'll make the changes and amend the PR